peps/pep-9999.rst

```diff
@@ -0,0 +1,632 @@
+prefix
+[`Issue 46 <https://github.com/capi-workgroup/problems/issues/46>`__].
+
+Some API function do not have the same behaviour as their Python
+equivalents.  The behaviour of ``PyIter_Next`` is different from
+``tp_iternext``.
+[`Issue 29 <https://github.com/capi-workgroup/problems/issues/29>`__].
+The behaviour of ``PySet_Contains`` is different from ``set.__contains__``
+[`Issue 6 <https://github.com/capi-workgroup/problems/issues/6>`__].
+
+The fact that ``PyArg_ParseTupleAndKeywords`` takes a non-const
+char* array as argument makes it more difficult to use.
+[`Issue 28 <https://github.com/capi-workgroup/problems/issues/28>`__].
+
+Python.h does not expose the whole API. Some headers (like marshal.h)
+are not included from Python.h.
+[`Issue 43 <https://github.com/capi-workgroup/problems/issues/43>`__].
+
+
+Missing Functionality
+---------------------
+
+This section consists of a list of feature requests, i.e., functionality
+that was identified as missing in the current C API.
+
+Debug Mode
+~~~~~~~~~~
+
+A debug mode that can be activated without recompilation and which
+activates various checks that can help detect various types of errors.
+[`Issue 36 <https://github.com/capi-workgroup/problems/issues/36>`__].
+
+Introspection
+~~~~~~~~~~~~~
+
+There aren't currently reliable introspection capabilities for objects
+defined in C in the same way as there are for Python objects.
+[`Issue 32 <https://github.com/capi-workgroup/problems/issues/32>`__].
+
+Efficient type checking for heap types, similar to what ``Py*_Check``
+can do for a static type.
+[`Issue 17 <https://github.com/capi-workgroup/problems/issues/17>`__].
+
+Improved Interaction with Other Languages
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Interfacing with other GC based languages, and integrating their
+GC with Python's GC.
+[`Issue 19 <https://github.com/capi-workgroup/problems/issues/19>`__].
+
+Inject foreign stack frames to the traceback.
+[`Issue 18 <https://github.com/capi-workgroup/problems/issues/18>`__].
+
+Concrete strings that can be used in other languages
+[`Issue 16 <https://github.com/capi-workgroup/problems/issues/16>`__].
+
+References
+==========
+
+1. `Python/C API Reference Manual <https://docs.python.org/3/c-api/index.html>`__
+2. `2023 Language Summit Blog Post: Three Talks on the C API <https://pyfound.blogspot.com/2023/05/the-python-language-summit-2023-three.html>`__
+3. `capi-workgroup on GitHub <https://github.com/capi-workgroup>`__
+4. `Irit's Discord post with Core Sprint 2023 slides about C API workgroup <https://discord.com/channels/854719841091715092/897862918115315813/1160952205801689249>`__
+5. `Petr's Discord post with Core Sprint 2023 slides <https://discord.com/channels/854719841091715092/897862918115315813/1160939833473126411>`__
+6. `Steve's Discord post with Core Sprint 2023 slides for Things to Learn from HPy <https://discord.com/channels/854719841091715092/897862918115315813/1160942893696024578>`__
```

Comment:
The Discord is private to the core dev team. Similar to [here](https://devguide.python.org/core-developers/become-core-developer/#gaining-commit-privileges), shall we denote that with a padlock symbol:

```suggestion
4. 🔒 `Irit's Discord post with Core Sprint 2023 slides about C API workgroup <https://discord.com/channels/854719841091715092/897862918115315813/1160952205801689249>`__
5. 🔒 `Petr's Discord post with Core Sprint 2023 slides <https://discord.com/channels/854719841091715092/897862918115315813/1160939833473126411>`__
6. 🔒 `Steve's Discord post with Core Sprint 2023 slides for Things to Learn from HPy <https://discord.com/channels/854719841091715092/897862918115315813/1160942893696024578>`__
```